shared/js/background/popupMessaging.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @typedef {import('webextension-polyfill').Runtime.Port} Port
+ * @typedef {import('@duckduckgo/privacy-dashboard/schema/__generated__/schema.types').IncomingExtensionMessage} OutgoingPopupMessage
+ */
+
+// Messaging connection with the popup UI (when active).
+/** @type {Port?} */
+let activePort = null
+
+/**
+ * Set up the messaging connection with the popup UI.
+ *
+ * Note: If the ServiceWorker dies while there is an open connection, the popup
+ *       will take care of re-opening the connection.
+ *
+ * @param {Port} port
+ * @param {Record<
+ *     string,
+ *     (options: any, sender: any, message: any) => any
+ * >} messageHandlers
+ */
+export function popupConnectionOpened (port, messageHandlers) {
+    activePort = port
```

Comment:
Is it possible at all to open two popups concurrently?

I just tested in Chrome and if I inspect a popup in one window, then I can open a second window and open a second popup at the same time. I wonder what will happen here? Will we just post data to the most recently opened, or are they going to fight over reconnects?

Reply:
In that situation, I think incoming (popup to extension) messages will be received OK, but replies and outgoing messages will be sent to the second popup. The disconnect listeners also take care to check that they really are the active connection, before clearing the active connection variable.

Just tested and it behaved like I expected, somewhat broken in that disabling protections in the first popup results in the protections being disabled for the second popup - but nothing blows up that I can see.

_(This is an extreme edge-case to worry about, I doubt a user would trigger it + our popup UI often breaks in these kinds of situations already when the current tab can't be determined.)_